### PR TITLE
Protection From Missing HLTVtx / PathResolver

### PR DESCRIPTION
**Pull request by johnda102**

### CODE REVIEW

File: Root/HLTJetRoIBuilder.cxx [hunk 1]
```diff
@@ -44,12 +44,13 @@ using std::vector;
 ClassImp(HLTJetRoIBuilder)
 
 HLTJetRoIBuilder :: HLTJetRoIBuilder () :
-  Algorithm("HLTJetRoIBuilder"),
+Algorithm("HLTJetRoIBuilder"),
```
Comment (kratsg) [inline Root/HLTJetRoIBuilder.cxx]: double-check the space here

File: Root/HLTJetRoIBuilder.cxx [hunk 2]
```diff
@@ -311,17 +316,19 @@ EL::StatusCode HLTJetRoIBuilder :: buildHLTBJets ()
...
-      cout << "ERROR Problem in container size: " << m_name
-	   << " jets: "<< jetCollections.size() << " " << m_jetName
-	   << " vtx: "<< vtxCollections.size()  << " " << m_vtxName << endl;
-      for ( unsigned ifeat=0 ; ifeat<vtxCollections.size() ; ifeat++ ) {
-	cout << "Feture: " << ifeat << endl;
-	for( auto vtx_itr : *(vtxCollections.at(ifeat).cptr()) ){
-	  cout << vtx_itr->vertexType() <<endl;
+    if(m_readHLTVtx){
+      if(vtxCollections.size() < jetCollections.size()){
+	cout << "ERROR Problem in container size: " << m_name
```
Comment (kratsg) [inline Root/HLTJetRoIBuilder.cxx]: use ATH_MSG_XXXX macros.

File: Root/HLTJetRoIBuilder.cxx [hunk 3]
```diff
@@ -311,17 +316,19 @@ EL::StatusCode HLTJetRoIBuilder :: buildHLTBJets ()
...
+	     << " jets: "<< jetCollections.size() << " " << m_jetName
+	     << " vtx: "<< vtxCollections.size()  << " " << m_vtxName << endl;
+	for ( unsigned ifeat=0 ; ifeat<vtxCollections.size() ; ifeat++ ) {
+	  cout << "Feture: " << ifeat << endl;
```
Comment (kratsg) [inline Root/HLTJetRoIBuilder.cxx]: use ATH_MSG_XXXX macros.

File: Root/HLTJetRoIBuilder.cxx [hunk 4]
```diff
@@ -311,17 +316,19 @@ EL::StatusCode HLTJetRoIBuilder :: buildHLTBJets ()
...
+	  for( auto vtx_itr : *(vtxCollections.at(ifeat).cptr()) ){
+	    cout << vtx_itr->vertexType() <<endl;
```
Comment (kratsg) [inline Root/HLTJetRoIBuilder.cxx]: use ATH_MSG_XXXX macros.

File: Root/HLTJetRoIBuilder.cxx [hunk 5]
```diff
@@ -409,53 +416,59 @@ EL::StatusCode HLTJetRoIBuilder :: buildHLTBJets ()
...
+	  //
+	  if(backupVtxCollections.size()){
+	    ATH_MSG_DEBUG("Have EFHistoPrmVtx.  ");
+	    m_vtx_hadDummyPV  (*newHLTBJet)         = '1';
+	    const xAOD::Vertex *backup_pvx = HelperFunctions::getPrimaryVertex(backupVtxCollections.at(ifeat).cptr(), msg());
+	    ATH_MSG_DEBUG("backup_pvx.  " << backup_pvx);
+	    m_vtx_decoration  (*newHLTBJet)         = backup_pvx;
+	    m_vtx_decoration_bkg(*newHLTBJet)       = backup_pvx;
+	  }else{
+	    cout << "No EFHistoPrmVtx....  " << endl;
```
Comment (kratsg) [inline Root/HLTJetRoIBuilder.cxx]: use ATH_MSG_XXXX macros.

**Comment by johnda102** [comment]
This commit doesnt introduce any couts. 
I will migrate toward the new messaging but I think this should be done all at once for the whole package

**Comment by kratsg** [comment]
> I will migrate toward the new messaging but I think this should be done all at once for the whole package

Oh, I didn't realize. I'll approve the changes now. I think I missed a few here and there to be honest :(

**Pull request merged**